openeo/udf/run_code.py

```diff
@@ -196,6 +195,37 @@ def run_udf_code(code: str, data: UdfData) -> UdfData:
             result_cube: xarray.DataArray = func(cube=data.get_datacube_list()[0].get_array(), context=data.user_context)
             data.set_datacube_list([XarrayDataCube(result_cube)])
             return data
+        elif (
+            fn_name in ["apply_vectorcube"]
+            and "geometries" in params
+            and "cube" in params
+            and _annotation_is_data_array(params["cube"].annotation)
+        ):
+            if data.get_feature_collection_list is None or data.get_datacube_list() is None:
+                raise ValueError(
+                    "The provided UDF expects a FeatureCollection and a datacube, but received {f} and {c}".format(
+                        f=data.get_feature_collection_list(), c=data.get_datacube_list()
+                    )
+                )
+            if len(data.get_feature_collection_list()) != 1:
+                raise ValueError(
+                    "The provided UDF expects exactly one FeatureCollection, but {c} were provided.".format(
+                        c=len(data.get_feature_collection_list())
+                    )
+                )
+            if len(data.get_datacube_list()) != 1:
+                raise ValueError(
+                    "The provided UDF expects exactly one datacube, but {c} were provided.".format(
+                        c=len(data.get_datacube_list())
+                    )
+                )
+            # TODO: geopandas is optional dependency.
```

Comment:
> TODO: geopandas is optional dependency

what does this refer to?

Reply:
It refers to the fact that the annotations check for geometries requires geopandas as a dependency. Because it is a geopandas geodataframe.
 
https://github.com/Open-EO/openeo-python-client/blob/58f0205db63efbc860b99477cc5e5dea6069224b/openeo/udf/feature_collection.py#L13

Reply:
note that you can also do the annotation check for geopandas through a string representation, without actually depending on `import geopandas`: e.g. see the annotation checks for `openeo_udf.api.datacube.DataCube`: which refers to the legacy `openeo_udf` project

Reply:
I believe that checking for an annotation string for an external library might be too brittle. Especially since users could write the annotation as for example "gpd.GeoDataFrame". I have tested it and `_get_annotation_str` will not match it in those cases.

Reply:
```python
import geopandas as gpd
import inspect

def foo(bar: gpd.GeoDataFrame):
    ...

for name, param in inspect.signature(foo).parameters.items():
    print(name, param.annotation)
```
gives 
```
bar <class 'geopandas.geodataframe.GeoDataFrame'>
```
So the annotation does not depend on how geopandas was imported

Reply:
I added an annotation check for the geometries parameter.